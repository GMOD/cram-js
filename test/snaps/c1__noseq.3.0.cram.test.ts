import { testFile } from '../testUtil'
testFile('c1#noseq.3.0.cram')
