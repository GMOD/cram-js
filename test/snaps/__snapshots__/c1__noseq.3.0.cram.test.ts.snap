// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole c1#noseq.3.0.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 44,
      "_size": 18,
      "alignmentSpan": 0,
      "crc32": 3519680774,
      "landmarks": [
        0,
        119,
      ],
      "length": 187,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 44,
        "_size": 18,
        "alignmentSpan": 0,
        "crc32": 3519680774,
        "landmarks": [
          0,
          119,
        ],
        "length": 187,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "c1",
              },
              {
                "tag": "LN",
                "value": "10",
              },
              {
                "tag": "M5",
                "value": "e224bd8a70e5466c6a033be886045c85",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/c1.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "c1",
              },
              {
                "tag": "LN",
                "value": "10",
              },
              {
                "tag": "M5",
                "value": "e224bd8a70e5466c6a033be886045c85",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/c1.fa",
              },
            ],
            "tag": "SQ",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": true,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "BA": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              2,
              2,
              2,
              2,
            ],
            "numCodes": 4,
            "numLengths": 4,
            "symbols": [
              65,
              67,
              71,
              84,
            ],
          },
          "parametersBytes": 10,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              0,
              4,
            ],
          },
          "parametersBytes": 6,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              3,
              11,
            ],
          },
          "parametersBytes": 6,
        },
        "DL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "FC": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              68,
            ],
          },
          "parametersBytes": 4,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              0,
              1,
            ],
          },
          "parametersBytes": 6,
        },
        "FP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              5,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              2,
              2,
            ],
            "numCodes": 3,
            "numLengths": 3,
            "symbols": [
              9,
              0,
              10,
            ],
          },
          "parametersBytes": 8,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              0,
              1,
            ],
          },
          "parametersBytes": 6,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {
        "MDZ": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 9,
            "stopByte": 9,
          },
          "parametersBytes": 2,
        },
        "NMC": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 3,
              "parameters": {
                "bitLengths": [
                  0,
                ],
                "numCodes": 1,
                "numLengths": 1,
                "symbols": [
                  1,
                ],
              },
              "parametersBytes": 4,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 1,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 9,
        },
      },
      "tagIdsDictionary": [
        [],
        [
          "MDZ",
          "NMC",
        ],
      ],
    },
    "containerHeader": {
      "_endPosition": 249,
      "_size": 18,
      "alignmentSpan": 10,
      "crc32": 2503485671,
      "landmarks": [
        260,
      ],
      "length": 510,
      "numBases": 56,
      "numBlocks": 7,
      "numLandmarks": 1,
      "numRecords": 9,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 249,
        "_size": 18,
        "alignmentSpan": 10,
        "crc32": 2503485671,
        "landmarks": [
          260,
        ],
        "length": 510,
        "numBases": 56,
        "numBlocks": 7,
        "numLandmarks": 1,
        "numRecords": 9,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "sq1",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 515,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "sQ1",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 516,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 11,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 10,
            "readName": "SQ1",
            "sequenceId": 0,
            "tags": {
              "MD": "10",
              "NM": 0,
            },
            "templateSize": 0,
            "uniqueId": 517,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "D",
                "data": 1,
                "pos": 5,
                "refPos": 5,
              },
            ],
            "readGroupId": -1,
            "readLength": 9,
            "readName": "sq2",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 518,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "D",
                "data": 1,
                "pos": 5,
                "refPos": 5,
              },
            ],
            "readGroupId": -1,
            "readLength": 9,
            "readName": "sQ2",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 519,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 11,
            "flags": 0,
            "lengthOnRef": 10,
            "mappingQuality": 0,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "D",
                "data": 1,
                "pos": 5,
                "refPos": 5,
              },
            ],
            "readGroupId": -1,
            "readLength": 9,
            "readName": "SQ2",
            "sequenceId": 0,
            "tags": {
              "MD": "4^G5",
              "NM": 1,
            },
            "templateSize": 0,
            "uniqueId": 520,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": "AACCCGGTT",
            "readGroupId": -1,
            "readLength": 9,
            "readName": "sq3",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 521,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": [
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
              255,
            ],
            "readBases": "AACCCGGTT",
            "readGroupId": -1,
            "readLength": 9,
            "readName": "sQ3",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 522,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 11,
            "flags": 4,
            "lengthOnRef": undefined,
            "mappingQuality": undefined,
            "qualityScores": null,
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 0,
            "readName": "SQ3",
            "sequenceId": 0,
            "tags": {},
            "templateSize": 0,
            "uniqueId": 523,
          },
        ],
        "header": {
          "_endPosition": 550,
          "_size": 36,
          "compressedSize": 32,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            10,
            9,
            0,
            5,
            4,
            1,
            9,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            226,
            36,
            189,
            138,
            112,
            229,
            70,
            108,
            106,
            3,
            59,
            232,
            134,
            4,
            92,
            133,
          ],
          "contentId": 0,
          "contentPosition": 514,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 2975191832,
          "parsedContent": {
            "_endPosition": 281,
            "_size": 32,
            "contentIds": [
              1,
              9,
              11,
              12,
            ],
            "md5": [
              226,
              36,
              189,
              138,
              112,
              229,
              70,
              108,
              106,
              3,
              59,
              232,
              134,
              4,
              92,
              133,
            ],
            "numBlocks": 5,
            "numContentIds": 4,
            "numRecords": 9,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 10,
            "refSeqStart": 1,
          },
          "uncompressedSize": 32,
        },
      },
    ],
  },
]
`;
